### ISSUE TITLE
Cannot find language for highlighting, but tests succeed

### ISSUE THREAD

**Issue by bollian** [tree-sitter/tree-sitter#1324]
Hello, I've created a tree-sitter parser for the OpenSCAD language at <https://github.com/bollian/tree-sitter-openscad>, and I've started working on syntax highlighting, but I can't get the `highlight` command to work. All the tests pass, but every time I run `npx tree-sitter highlight examples/*`, I get a bunch of errors saying `No langauge found for path "examples/basic.scad"` for all the files I provide the command.

I'm fairly certain I've setup the tree-sitter config properly. My parser is stored at '/home/bollian/code/tree-sitter/tree-sitter-openscad', and my '~/.tree-sitter/config.json' looks like this:

```
{
  "parser-directories": [
    "/home/bollian/code/tree-sitter"
  ],
  "theme": { /* omitted for brevity */ }
}
```

The 'package.json' file for my parser is specifying the filename extension to be used to identify OpenSCAD files with this top-level entry:

```
"tree-sitter": {
  "scope": "source.openscad",
  "file-types": [
    "scad"
  ]
}
```

So why can't the tree-sitter command recognize the language of my files? What am I doing wrong?

**Comment by ahlinc** [comment]
Try to move your config to the `~/.config/tree-sitter/config.json` path and then fing your language in the output of the `tree-sitter dump-languages` command.

**Comment by bollian** [comment]
I moved the config and the result was the same. `npx tree-sitter dump-languages` shows no output.

**Comment by ahlinc** [comment]
Could you show an output for the `tree-sitter -V`?

**Comment by ahlinc** [comment]
The `tree-sitter` key requires a list, that's the cause of the issue.
```diff
diff --git i/package.json w/package.json
index 607254e..880b2c0 100644
--- i/package.json
+++ w/package.json
@@ -21,10 +21,12 @@
   "devDependencies": {
     "tree-sitter-cli": "^0.19.5"
   },
-  "tree-sitter": {
-    "scope": "source.openscad",
-    "file-types": [
-      "scad"
-    ]
-  }
+  "tree-sitter": [
+    {
+      "scope": "source.openscad",
+      "file-types": [
+        "scad"
+      ]
+    }
+  ]
 }
```

```console
tree-sitter-openscad # ts dump-languages | grep -B1 -A5 openscad

scope: source.openscad
parser: "/home/user1/Code/tree-sitter-openscad/"
highlights: None
file_types: ["scad"]
content_regex: None
injection_regex: None
```

**Comment by ahlinc** [comment]
Also be aware of #1318, it seems on your grammar that also happens.

**Comment by bollian** [comment]
I changed the `tree-sitter` key to a list, but am still seeing the same error, and `dump-languages` still lists nothing.

```
> npx tree-sitter -V
tree-sitter 0.19.5 (8d8690538ef0029885c7ef1f163b0e32f256a5aa)
```
I'm intentionally using 0.19.5 because I'm hoping to use this parser from rust, and the tree-sitter crate on "crates.io" hasn't hit 0.20 yet. When I try using `tree-sitter-cli` 0.20, I'm unable to build the rust package and I get other mysterious highlighting errors:

```
> npx tree-sitter highlight examples/*
thread 'main' panicked at 'index out of bounds: the len is 2 but the index is 16', cli/src/highlight.rs:359:34
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by bollian** [comment]
Yeah, that's how I first set up the config. I've tried it both ways since then, and neither configuration works. For context, my grammar folder is '/home/bollian/code/tree-sitter/tree-sitter-openscad' and my `config.json` has '/home/bollian/code/tree-sitter' as its only listed parser directory.

**Comment by bollian** [comment]
So today I went through and reinstalled npm and all dependencies, regenerated the config, and cleared the tree-sitter cache and the issue went away. Unfortunately, I don't know what the exact cause was because I can't reproduce it now. Maybe there was something funky in the cache since I switched multiple times between 0.20 and 0.19? I hadn't known that the cache existed until today, so today was the first time I cleared it.

Anyway, thank you for all your help. Since I can't reproduce any more, I'm closing the issue.
